app/grandchallenge/ai_website/views.py

```diff
@@ -0,0 +1,187 @@
+                "subspecialities": subspecialities,
+                "modalities": modalities,
+                "selected_subspeciality": subspeciality_query,
+                "selected_modality": modality_query,
+                "products_selected_page": True,
+                "product_total": len(self.queryset),
+            }
+        )
+        return context
+
+
+class ProductPage(TemplateView):
+    template_name = "ai_website/product_page.html"
+
+    def get_context_data(self, pk):
+        product = get_object_or_404(ProductEntry, pk=pk)
+        return {"product": product}
+
+
+class CompanyList(ListView):
+    template_name = "ai_website/company_list.html"
+    model = CompanyEntry
+    context_object_name = "companies"
+    queryset = CompanyEntry.objects.order_by("company_name")
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        search_query = self.request.GET.get("search")
+
+        if search_query:
+            search_fields = ["company_name", "description", "hq"]
+            q = reduce(
+                or_,
+                [
+                    Q(**{f"{f}__icontains": search_query})
+                    for f in search_fields
+                ],
+                Q(),
+            )
+            queryset = queryset.filter(q)
+        return queryset
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        search_query = self.request.GET.get("search", "")
+
+        context.update(
+            {
+                "q_search": search_query,
+                "companies_selected_page": True,
+                "company_total": len(self.queryset),
+            }
+        )
+        return context
+
+
+class CompanyPage(TemplateView):
+    template_name = "ai_website/company_page.html"
+
+    def get_context_data(self, pk):
+        company = get_object_or_404(CompanyEntry, pk=pk)
+        products_by_company = ProductEntry.objects.filter(
+            company__company_name__contains=company.company_name
+        ).order_by("product_name")
+        return {"company": company, "products_by_company": products_by_company}
+
+
+class AboutPage(TemplateView):
+    template_name = "ai_website/about.html"
+
+
+class ContactPage(TemplateView):
+    template_name = "ai_website/contact.html"
+
+
+class ImportDataView(LoginRequiredMixin, PermissionRequiredMixin, FormView):
+    template_name = "ai_website/import_data.html"
+    form_class = ImportForm
+    permission_required = (
+        f"{ProductEntry._meta.app_label}.add_{ProductEntry._meta.model_name}"
+    )
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update({"user": self.request.user})
+        return kwargs
+
+    def form_valid(self, *args, **kwargs):
+        response = super().form_valid(*args, **kwargs)
+        form = self.get_form()
+        if form.is_valid():
+            di = DataImporter()
+            di.import_data(
```

Comment:
How long does this take?

Reply:
This takes a couple of seconds on my local machine with the current dataset.

Reply:
Ok, as long as it is not going to kill the view